src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js

```diff
@@ -347,46 +340,56 @@ class FieldBlocks extends React.Component<FieldTypeProps<Array<BlockEntry>>> {
     };
 
     @action handleSettingsClick = (index: number) => {
-        if (!this.settingsFormKey || !this.value) {
+        const settingsFormKey = this.settingsFormKey;
+
+        if (!settingsFormKey || !this.value) {
             return;
         }
 
-        this.blockSettingsOpen = index;
-        this.blockSettingsFormStore.setMultiple(this.value[index][SETTINGS_KEY] ?? {});
-        this.blockSettingsFormStore.dirty = false;
+        // create new formstore to make sure that overlay displays correct data
+        this.blockSettingsFormStore?.destroy();
+        this.blockSettingsFormStore = memoryFormStoreFactory.createFromFormKey(
+            settingsFormKey,
+            this.value[index][SETTINGS_KEY],
```

Comment:
Think here a deep copy would be needed that the `X` just close the overlay and does not apply its changes.

Reply:
Could solve it by doing:

```suggestion
            toJS(this.value[index][SETTINGS_KEY]),
```

not sure if we should use `toJS` in this case.

Reply:
fixed 🙂